includes/admin/reports/class-wc-admin-report-v2.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * V2 report functionality
+ *
+ * @package WooCommerce\Admin\Reports
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+if ( ! class_exists( 'WC_Admin_Report' ) ) {
+	require_once dirname( __FILE__ ) . '/class-wc-admin-report.php';
+}
+
+/**
+ * Class WC_Admin_Report_V2
+ * Makes use of the WC Admin plugin endpoints if it is not disabled.
+ */
+class WC_Admin_Report_V2 extends WC_Admin_Report {
+
+	/**
+	 * Overwrites the original sparkline to use the new reports data if WooAdmin is enabled.
+	 * Prepares a sparkline to show sales in the last X days.
+	 *
+	 * @param  int    $id ID of the product to show. Blank to get all orders.
+	 * @param  int    $days Days of stats to get.
+	 * @param  string $type Type of sparkline to get. Ignored if ID is not set.
+	 * @return string
+	 */
+	public function sales_sparkline( $id = '', $days = 7, $type = 'sales' ) {
+		$is_wc_admin_disabled = apply_filters( 'woocommerce_admin_disabled', false );
+		if ( $is_wc_admin_disabled ) {
+			return parent::sales_sparkline( $id, $days, $type );
```

Comment:
Uses original logic if the wc_admin is disabled.
Otherwise make use of the `revenue/stats` endpoint.